### PR TITLE
Fix bug: shifted unitcell boundaries

### PR DESCRIPTION
**Pull request by adrianusler**
It works again. Ready to merge, @ab5424
grafik

## Summary by CodeRabbit

- **New Features**
	- Introduced a `coordinate_shift` variable in the `Structure` class to enhance coordinate management.
  
- **Bug Fixes**
	- Improved the accuracy of unit cell boundary calculations in the `Crystal` class by updating the method for determining positions.

- **Tests**
	- Adjusted the `test_unitcell_boundaries_positions` to reflect changes in position calculations, ensuring tests align with the updated logic.

### CODE REVIEW

**Comment by ab5424** [comment]
@coderabbit review

File: src/molara/structure/structure.py [hunk 1]
```diff
@@ -88,6 +89,7 @@ def calculate_center_of_mass(self: Structure) -> np.ndarray:
     def center_coordinates(self: Structure) -> None:
         """Centers the structure around the center of mass."""
         self.center_of_mass = self.calculate_center_of_mass()
+        self.coordinate_shift += self.center_of_mass  # change to geometric center if needed
         self.geometric_center = np.mean(self.coords, axis=0)
```
[the remark this answers is not in the document]
Comment (adrianusler) [reply src/molara/structure/structure.py:93]: There is implementation uncertainty and I think that's ok. See issue

**Comment by ab5424** [comment]
@adrianusler Sorry to ask again but why is `geometric_center` calculated on line 93 and again on line 103? Is this required?

**Comment by adrianusler** [comment]
> @adrianusler Sorry to ask again but why is `geometric_center` calculated on line 93 and again on line 103? Is this required?

Well this is technically not a part of this Pull Request, but has already been accepted into the main branch. Currently, this is unnecessary. Should we change the recentering from centering on the geometric center, rather than the center of mass, it will be necessary.

**Comment by ab5424** [comment]
@adrianusler I see.

**Review by ab5424 — APPROVED** [review]

**Comment by adrianusler** [comment]
> @adrianusler I see.

I also think that this section of the code isn't the most elegant one, though @ab5424

**Pull request merged**